crates/sui-core/src/authority_server.rs

```diff
@@ -995,11 +995,25 @@ macro_rules! handle_with_decoration {
                     None
                 }
             }
-            ClientIdSource::XForwardedFor => {
+            ClientIdSource::XForwardedFor(num_hops) => {
                 let do_header_parse = |op: &MetadataValue<Ascii>| {
                     match op.to_str() {
                         Ok(header_val) => {
-                            match header_val.parse::<SocketAddr>() {
+                            let header_contents = header_val.split(',').map(str::trim).collect::<Vec<_>>();
+                            let contents_len = header_contents.len();
+                            if contents_len < num_hops + 1 {
```

Comment:
instead of checking this and then using `[]` below, can you use `get` and have the error return in the None case? I agree this code can't panic, but it would be nicer to make it very clear

Reply:
Done